### PR TITLE
Implementation of Function countItem

### PR DESCRIPTION
**Pull request by diegodsgarcia**
- Implementation of Function coutItem
- Execute the coutItem in a hard code 
- Remove not necessary coments

### CODE REVIEW

File: textalyze.js [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
-
 /**
  * Given an input Array, returns a Map containing the count of each item in the input.
  * @param {Array} array - The array of items to count
  * @param {Map} counts - A Map containing the counts of the items in the input array
  */
+
 function itemCounts(array) {
-  let counts = new Map();
+  const counts = new Map();
```
Comment (jfarmer) [inline textalyze.js]: Exercise: how would you write this as a single call to [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)?

File: textalyze.js [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+      counts.set(item, 1);
+    }
+  });
 
   return counts;
 }
 
 module.exports = { itemCounts };
+
+// Result
```
Comment (jfarmer) [inline textalyze.js]: We'd like this file to be both included as a module and run via `npm start`.  Check out https://stackoverflow.com/questions/4981891/node-js-equivalent-of-pythons-if-name-main for how to do this.

You want

```javascript
if (require.main == module) {
  const items = ['a', 'a', 'a', ...];
}
```

File: textalyze.js [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
+const items = ['a', 'a', 'a', 'b', 'b', 'c'];
+console.log(`The counts for [${items}] are...`); 
+
+itemCounts(items).forEach((value, key) => {
```
Comment (jfarmer) [inline textalyze.js]: Check out [console.table](https://developer.mozilla.org/en-US/docs/Web/API/Console/table) — it's useful for quickly printing out tables in debugging situations.

**Comment by jfarmer** [comment]
Some English feedback.  You wrote:

```
Implementation of Function coutItem
Execute the coutItem in a hard code
Remove not necessary coments
```

Capitalizing "Function" is strange, although you could be referring to the `Function` class.  The function's actual name is `itemCounts`, though.


```
Execute the coutItem in a hard code
```

You mean to say something like..."Run itemCounts with hard-coded input"

```
Remove not necessary coments
```

Should be: "Remove unnecessary comments".  Notice the two m's in "comments"

File: textalyze.js (hunk 3 above)
Comment (diegodsgarcia) [reply textalyze.js]: Thanks for tip! I see that console.table only works in array and object, right? For instances of Map(), I think this doesn't work right?

**Pull request merged**